### PR TITLE
VisitPush event type for Clinical Summary + some extensions

### PR DESCRIPTION
**Pull request by chennen-carbon**
In order to use Redox as a query responder on the Carequality network, we need to push Clinical Summary data to their Data-on-Demand product via the `PatientPush` and `VisitPush` event types:

https://developer.redoxengine.com/data-models/ClinicalSummary.html#PatientPush
https://developer.redoxengine.com/data-models/ClinicalSummary.html#VisitPush

The `PatientPush` event type is already present in this SDK but `VisitPush` needs to be added.

Additionally, in order to use Redox to query for patients/organizations/documents on the Carequality network, we need a few extensions for the Meta object in the query datamodels.

### CODE REVIEW

File: src/main/scala/com/github/vitalsoftware/scalaredox/models/clinicalsummary/VisitPush.scala [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+  ReasonForReferralText: Option[String] = None,
+  ReasonForVisitText: Option[String] = None,
+  Results: Seq[ChartResult] = Seq.empty,
+  ReviewOfSystemsText: Option[String] = None,
+  SocialHistory: Option[SocialHistory] = None,
+  SubjectiveText: Option[String] = None,
+  VitalSigns: Seq[VitalSigns] = Seq.empty
+) extends ClinicalSummaryLike
+
+object PatientPush extends RobustPrimitives
```
Comment (talha-sen-carbon) [inline src/main/scala/com/github/vitalsoftware/scalaredox/models/clinicalsummary/VisitPush.scala:83]: This needs to be VisitPush

**Review by talha-sen-carbon — CHANGES_REQUESTED** [review]
1 small change needed. Also, are you planning to update the version with this PR or not? If you plan to update it with this PR, you need to include the version change too. If not, thats okay. Though I think coupling the version change with the PR is better logically.

File: src/main/scala/com/github/vitalsoftware/scalaredox/models/clinicalsummary/VisitPush.scala (hunk 1 above)
Comment (talha-sen-carbon) [inline src/main/scala/com/github/vitalsoftware/scalaredox/models/clinicalsummary/VisitPush.scala:83]:
```suggestion
object VisitPush extends RobustPrimitives
```

**Review by atillacarbon — COMMENTED** [review]
Should we add a test class `VisitPushTest.scala` like `ClinicalSummaryTest.scala` to test if we have created the data structure correctly? Or how we gonna test it?

**Review by talha-sen-carbon — APPROVED** [review]

**Review by talha-sen-carbon — COMMENTED** [review]
Currently, VisitPush is the same as PatientPush. Though I don't think it is necessary, it would be a nice touch.

**Comment by chennen-carbon** [comment]
> Should we add a test class `VisitPushTest.scala` like `ClinicalSummaryTest.scala` to test if we have created the data structure correctly? Or how we gonna test it?

Nothing to test, really? It's just a case class with a regular JSON serializer.

**Comment by chennen-carbon** [comment]
> Currently, VisitPush is the same as PatientPush. Though I don't think it is necessary, it would be a nice touch.

I'm not sure what change you're suggesting? VisitPush has (slightly) different properties than PatientPush, and was copied from Redox's dev docs online.

**Comment by atillacarbon** [comment]
> > Should we add a test class `VisitPushTest.scala` like `ClinicalSummaryTest.scala` to test if we have created the data structure correctly? Or how we gonna test it?
> 
> Nothing to test, really? It's just a case class with a regular JSON serializer.

Gotcha! This is the opposite, we producing the json. I going to approve.
fyi: Some test failures.

**Review by atillacarbon — DISMISSED** [review]

**Review by talha-sen-carbon — APPROVED** [review]
lgtm

**Pull request merged**